src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFakeImpl.java

```diff
@@ -25,10 +28,33 @@ public void createUrlAlias(UrlAlias urlAlias) {
     return aliases.get(alias);
   }
 
+  /**
+   * Delete the UrlAlias if exists from aliases HashMap.
+   *
+   * @param email of the user to whom the UrlAlias belongs
+   * @param alias the UrlAlias which should be removed
+   *
+   * @throws IllegalArgumentException if no such email or alias
+   */
+
   @Override
   public void deleteUrlAlias(String email, String alias) {
-    // TODO: We should implement it
-    throw new UnsupportedOperationException();
+    UrlAlias foundUrlAlias = null;
+
+    for (UrlAlias urlAlias : aliases.values()) {
+      if (foundUrlAlias == null) {
+        if (urlAlias.email().equals(email) && urlAlias.alias().equals(alias)) {
+          foundUrlAlias = urlAlias;
+          break;
+        }
+      }
+    }
+    if (foundUrlAlias == null) {
+      throw new IllegalArgumentException();
```

Comment:
RuntimeException if the repository does not contain a URL alias with this short name.
```suggestion
      throw new RuntimeException();
```